client/package.json

```diff
@@ -832,6 +823,7 @@
   },
   "dependencies": {
     "find": "^0.2.7",
+    "language-server-bitbake": "^2.2.0",
```

Comment:
What is the motivation behind this change? Why is it needed?

Reply:
This is imported by client/src/language/server.ts

Reply:
I mean, why is it required to do this decoupling. Is it required to achieve something later?

Reply:
Oh, just because in https://github.com/yoctoproject/vscode-bitbake/pull/156#issuecomment-1985616109, you said

> where we copy the server compiled JS

> I agree that it's not pretty, but that's still the only way we could get this to work.

I want to try can we decouple it, that is without copy server code to client. In fact, I refer the following projects:

- [diagnostic-languageserver](https://github.com/iamcco/diagnostic-languageserver): server
- [coc-diagnostic](https://github.com/iamcco/coc-diagnostic): client, which run server in https://github.com/iamcco/coc-diagnostic/blob/master/src/server.ts , then call `out/server.js` in client https://github.com/iamcco/coc-diagnostic/blob/master/src/index.ts#L43

Reply:
Ok so this is for applying best practices, which is nice to have in itself. However, we have to make sure there are no regressions in the other configurations which I listed in another comment. I expect them to be quite challenging to implement though. However, if you find a solution that works for all of them, I'll gladly merge the result!

If you are determined to improve the decoupling, you could have a look [at some official vscode client/language servers](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#lsp-sample-a-simple-language-server-for-plain-text-files). The example we are based on is quite old and deprecated compared to this one.

Reply:
> The example we are based on is quite old and deprecated compared to this one.

Yes, is any plan to upgrade it?

> be quite challenging to implement though

For me, it is a little challenging to figure out why some feature break really. Now we still have 3 tests failed.

Reply:
Do you know why the 3 tests failed?

Reply:
Try rebasing on staging, we had an unfortunate rebase which broke at least one test. Otherwise I guess integration tests migh fail because there is a disynchronisation between the npm language server and the latest staging commits because of your change. The other debug configurations, you'll have to try out yourself.

All of those things are because your change uses the server from npm rather than the sources. As I said, it's going to be tricky if you want to fix this while keeping your change.

Reply:
> All of those things are because your change uses the server from npm rather than the sources

You are right. Now all thing work.